app/src/main/java/com/example/dine_and_donate/UpcomingVouchersFragment.java

```diff
@@ -32,50 +34,84 @@
     private User mCurrUser;
     private DatabaseReference mRef;
     private DatabaseReference mRefForEvent;
+    private View mView;
+    private TextView mEmptyView;
+    private int mSize;
+    private StaggeredRecyclerViewAdapter mStaggeredRecyclerViewAdapter;
     private ArrayList<Event> mEvents = new ArrayList<>();
 
     //create view based on data in array lists, inflates the layout of the fragment
     public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
-        return inflater.inflate(R.layout.tab_fragment, container, false);
+        HomeActivity homeActivity = (HomeActivity) getActivity();
+        mCurrUser = homeActivity.currentUser;
+
+        mView = inflater.inflate(R.layout.tab_fragment, container, false);
+        mRecyclerView = mView.findViewById(R.id.rv_vouchers);
+        mEmptyView = mView.findViewById(R.id.empty_view);
+        mStaggeredRecyclerViewAdapter = new StaggeredRecyclerViewAdapter(getActivity(), mEvents);
+        StaggeredGridLayoutManager staggeredGridLayoutManager = new StaggeredGridLayoutManager(2, LinearLayoutManager.VERTICAL);
+        mRecyclerView.setLayoutManager(staggeredGridLayoutManager);
+        mStaggeredRecyclerViewAdapter.notifyDataSetChanged();
+        mRecyclerView.setAdapter(mStaggeredRecyclerViewAdapter);
+        return mView;
     }
 
+    //case 1: opens on profile when user first opens app (adapter = 0)
+    //case 2: switch from profile to map then back to map (adapter stays the same)
+    //case 3: save a new event on map and then go back to profile
     @Override
-    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
-        super.onViewCreated(view, savedInstanceState);
+    public void onViewCreated(@NonNull final View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(mView, savedInstanceState);
+//        int prevSize = mSavedEventsIDs.size();
+        Log.d("prevSize", mSize + "");
+//        Log.d("currSize", mSavedEventsIDs.size() + "");
+
+        //for the first
+        if(mStaggeredRecyclerViewAdapter.getItemCount() == 0) {
+            mEmptyView.setVisibility(View.GONE);
+            loadVouchers();
+        } else {
+            mEmptyView.setVisibility(View.VISIBLE);
+        }
+
+    }
+
+    //add images and descriptions to respective arrayLists
+    private void initBitmapsUpcomingEvents(String mUrls, String mDescriptions) {
```

Comment:
this class is duplicated right? it's probably best to put it in a helper class and reuse it in both fragments